sklearn/preprocessing/label.py

```diff
@@ -500,46 +547,51 @@ def label_binarize(y, classes, neg_label=0, pos_label=1, sparse_output=False):
         y_in_classes = in1d(y, classes)
         y_seen = y[y_in_classes]
         indices = np.searchsorted(sorted_class, y_seen)
-        indptr = np.hstack((0, np.cumsum(y_in_classes)))
+        if pos_label != 0:
+            indptr = np.hstack((0, np.cumsum(y_in_classes)))
+            data = np.empty_like(indices)
+            data.fill(pos_label)
+            y_out = sp.csr_matrix((data, indices, indptr),
+                                  shape=(n_samples, n_classes))
+        else:
+            indptr = np.arange(len(y_in_classes))
+            data = np.empty((n_samples, n_classes))
+            data.fill(neg_label)
+            data[indptr, indices] = 0
+            y_out = sp.csr_matrix(data)
 
-        data = np.empty_like(indices)
-        data.fill(pos_label)
-        Y = sp.csr_matrix((data, indices, indptr),
-                          shape=(n_samples, n_classes))
     elif y_type == "multilabel-indicator":
-        Y = sp.csr_matrix(y)
+        y_out = sp.csr_matrix(y)
```

Comment:
ok, I was just following the pep8 whete variables should be low-case, right?
and even before it was not always matrix in the final form like L485